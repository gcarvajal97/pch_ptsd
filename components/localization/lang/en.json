{
    "LearnMore": {
        "titleNotAlone": "You Are Not Alone",
        "titleHowLong": "How Long Do Traumatic Stress Reactions Last?"
    }
}