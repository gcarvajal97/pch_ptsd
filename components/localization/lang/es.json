{
    "LearnMore": {
        "titleNotAlone": "No Estas Solo",
        "titleHowLong": "¿Cuánto Duran Las Reacciones de Estrés Traumático?"
    }
}