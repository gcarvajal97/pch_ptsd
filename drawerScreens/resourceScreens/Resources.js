import React from "react";
import { StyleSheet, Text, View } from "react-native";
import { Linking } from "expo";
import colors from "../../colors";

function Resources(content) {
    if(content && content.length) {
        return (
            <View>
                {content.map((item, index) => {
                    return (
                        <View style={styles.contentContainer} key={index}>
                            <Text
                                style={styles.link}
                                onPress={() => Linking.openURL(item.link.url)}
                                accessible
                                accessibilityLabel={item.link.accessibilityLabel}
                                accessibilityHint={item.link.accessibilityHint}
                            >
                                {item.link.text}
                            </Text>
                            <Text style={styles.paragraph}>{item.contentText}</Text>
                        </View>
                    );
                })}
            </View>
        );
    } else {
        return (
            <View>
                <Text>Error: Resources content is in valid!</Text>
            </View>
        );
    }
}

export default Resources;

const styles = StyleSheet.create({
    contentContainer: {
        padding: 16,
        paddingVertical: 12,
        backgroundColor: colors.primary.white
    },
    paragraph: {
        fontSize: 14,
    },
    link: {
        fontSize: 16,
        fontFamily: "avenir-heavy",
        color: colors.accent,
        textDecorationLine: "underline",
    }
});
