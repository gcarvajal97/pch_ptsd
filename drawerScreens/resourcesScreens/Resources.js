import React from "react";
import { StyleSheet, Text, View } from "react-native";
import { Linking } from "expo";

function Resources(content) {
    content = Array.from(content);
    return (
        <View>
            {content.map((item, index) => {
                return (
                    <View style={styles.contentContainer} key={index}>
                        <Text
                            style={styles.link}
                            onPress={() => Linking.openURL(item.link.url)}
                            accessible
                            accessibilityLabel={item.link.accessibilityLabel}
                            accessibilityHint={item.link.accessibilityHint}
                        >
                            {item.link.text}
                        </Text>
                        <Text style={styles.paragraph}>{"\n"}{item.contentText}</Text>
                    </View>
                );
            })}
        </View>
    );
}

export default Resources;

const styles = StyleSheet.create({
    contentContainer: {
        padding: 20,
        backgroundColor: "white"
    },
    paragraph: {
        fontSize: 14,
        marginVertical: 2
    },
    link: {
        fontSize: 16,
        fontWeight: "bold",
        color: "#2089DC",
        textDecorationLine: "underline",
        marginVertical: 2
    }
});
