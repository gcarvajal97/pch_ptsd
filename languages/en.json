{
    "drawerNavigator": {
        "home": "Home", 
        "resources": "Resources", 
        "glossary": "Glossary" 
    },
    "mainScreen": {
        "pediatricPTSD": "Pediatric PTSD", 
        "more": "more",
        "learnButton": "LEARN MORE",
        "learnTitle":"Learn More About Injury and Trauma",
        "learnCaption": "Get up-to-date information and expert guidance to assist you and your injured child.",
        "learnAcessLabel": "Photo of a doctor holding a stethoscope with their arms folded accross their chest.",
        "learnAccessHint": "Navigates to the learn more screen",
        "quizButton": "TAKE THE QUIZ",
        "quizTitle": "Rate Your Child's Reactions",
        "quizCaption": "An injury or accident can be a scary or stressful experience. Take a quick quiz to measure your child's traumatic stress.",
        "quizAccessLabel": "Photo of child covering her eyes with her hands",
        "quizAccessHint": "Navigates to the quiz screen",
        "findHelpButton": "FIND HELP",
        "findHelpTitle": "Help Your Child Recover",
        "findHelpCaption": "Dealing with emotional reactions and injury care can be difficult. Find ways to help your child recover.",
        "findHelpAccessLabel": "Photo of father reading a book to a baby on his lap.",
        "findHelpAccessHint": "Navigates to the find help screen"
    },
    "learnMoreList": {
        "header": "Learn More",
        "title": "WHAT TO EXPECT AFTER INJURY",
        "listNames": [
            { 
                "name": "You Are Not Alone", 
                "navigateTo": "NotAlone", 
                "accessibilityHint": "Navigates to the you are not alone screen"
            },
            { 
                "name": "Reactions to Injury", 
                "navigateTo": "Reactions", 
                "accessibilityHint": "Navigates to the reactions to injury screen"
            },
            { 
                "name": "What Are Traumatic Stress Reactions?", 
                "navigateTo": "TraumaticStressReactions", 
                "accessibilityHint": "Navigates to the what are traumatic stress reactions screen"
            },
            { 
                "name": "How Long Do Traumatic Stress Reactions Last?",
                "navigateTo": "HowLong",
                "accessibilityHint": "Navigates to the how long do traumatic stress reactions last screen"
            }
        ]
    },
    "youAreNotAloneScreen": {
        "header": "Learn More",
        "title": "You Are Not Alone",
        "content": {
            "videoCard": {
                "accessibility": "Video on how injury is common",
                "boldText": "Your child is not alone.",
                "subtitle": "Every year, millions of children in the United States are injured."
            },
            "statsBullets": {
                "intro": "Each year in the United States, child injury statistics show that:",
                "bulletOne": "1 in 4 children and teens need medical care for an injury.",
                "bulletTwo": "7 million injured children are treated in the ER.",
                "bulletThree": "Several hundred thousand injured children are admitted to the hospital."
            },
            "paragaphOne": "Right now, hundreds of thousands of children across the country are recovering from an injury. Like you, their parents and other caring adults want to help them the best way possible. While it is important to tend to the wounds and rehabilitation, it is just as important to remember to look beyond the physical injuries.",
            "paragraphTwo": "With all the doctors, nurses, and therapists who will treat your child’s injury, remember that no one is more important than you for your child’s recovery. In making this application, we took the best information available from science and best practices to help you and your child recover after an injury."
        }
    }
}