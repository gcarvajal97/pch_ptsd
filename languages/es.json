{
    "drawerNavigator": {
        "home": "Hogar", 
        "resources": "Recursos", 
        "glossary": "Glosario"
    },
    "mainScreen": {
        "pediatricPTSD": "Pediátrico TEPT", 
        "more": "más",
        "learnButton": "APRENDE MÁS",
        "learnTitle":"Aprenda Más Sobre Lesiones y Traumas",
        "learnCaption": "Obtenga información actualizada y orientación experta para ayudarlo a usted y a su hijo lesionado.",
        "learnAcessLabel": "Foto de un médico sosteniendo un estetoscopio con los brazos cruzados sobre el pecho.",
        "learnAccessHint": "Navega a la pantalla para obtener más información.",
        "quizButton": "RESPONDER AL CUESTIONARIO",
        "findHelpButton": "ENCONTRAR AYUDA"
    }
}