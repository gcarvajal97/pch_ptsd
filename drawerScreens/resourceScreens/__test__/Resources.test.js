import React from 'react';
import Resources from '../Resources';
import renderer from 'react-test-renderer';
//import { Text } from "react-native-elements";
import { Linking } from "expo";
import { StyleSheet, Text, View } from "react-native";
import Enzyme, { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import en from '../../../languages/en.json';
import es from '../../../languages/es.json';

configure({ adapter: new Adapter() });
jest.mock('expo', () => ({
    Linking: {
        openURL: jest.fn()
    }
}));

describe("Resources", () => {

    afterEach(() => {
        jest.clearAllMocks();
    });

    it("clicks external link, opens correct website", () => {
        const translation = JSON.stringify(en.resources.accordion);
        const content = [translation];
        const result = Resources(content);      
        const wrapper = shallow(result);    
        wrapper.find('Text').first().props().onPress();
    })

    it('renders properly in english', () => {
        const translation = JSON.stringify(en.resources.accordion);
        const content = [translation];
        const result = Resources(content);
        expect(result).toMatchSnapshot();
    });

    it('renders properly in spanish', () => {
        const translation = JSON.stringify(es.resources.accordion);
        const content = [translation];
        const result = Resources(content);
        expect(result).toMatchSnapshot();
    });

    it('renders one resource', () => {
        const content = [  
            {
                link: {
                    text: "Resource Title",
                    url: "http://www.website.org",
                    accessibilityHint: "An accessiblity hint",
                    accessibilityLabel: "An accessibility label"
                },
                "contentText": "Resource description"
            }
        ];
        const result = Resources(content);
        expect(result).toMatchSnapshot();
    });

    it('renders two resources', () => {
        const content = [  
            {
                link: {
                    text: "Resource Title 1",
                    url: "http://www.website1.org",
                    accessibilityHint: "An Accessiblity Hint 1",
                    accessibilityLabel: "An Accessibility Label 1"
                },
                "contentText": "Resource Description 1"
            },
            {
                link: {
                    text: "Resource Title 2",
                    url: "http://www.website2.org",
                    accessibilityHint: "An accessiblity hint 2",
                    accessibilityLabel: "An accessibility label 2"
                },
                "contentText": "Resource Description 2"
            }
        ];
        const result = Resources(content);
        expect(result).toMatchSnapshot();
    });

    it('renders error message if resources are missing', () => {
        const content = [];
        const result = Resources(content);
        expect(result).toMatchSnapshot();
    });

    it('renders error message if resources are not defined', () => {
        const result = Resources(undefined);
        expect(result).toMatchSnapshot();
    });
});

const styles = StyleSheet.create({
    link: {
        fontSize: 16,
        fontWeight: "bold",
        color: "#2089DC",
        textDecorationLine: "underline",
        marginVertical: 2
    }
});