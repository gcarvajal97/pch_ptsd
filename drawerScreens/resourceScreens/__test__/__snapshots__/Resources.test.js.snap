// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resources renders error message if resources are missing 1`] = `
<View>
  <Text>
    Error: Resources content is in valid!
  </Text>
</View>
`;

exports[`Resources renders error message if resources are not defined 1`] = `
<View>
  <Text>
    Error: Resources content is in valid!
  </Text>
</View>
`;

exports[`Resources renders one resource 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "padding": 20,
      }
    }
  >
    <Text
      accessibilityHint="An accessiblity hint"
      accessibilityLabel="An accessibility label"
      accessible={true}
      onPress={[Function]}
      style={
        Object {
          "color": "#2089DC",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginVertical": 2,
          "textDecorationLine": "underline",
        }
      }
    >
      Resource Title
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
          "marginVertical": 2,
        }
      }
    >
      

      Resource description
    </Text>
  </View>
</View>
`;

exports[`Resources renders properly in english 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "padding": 20,
      }
    }
  >
    <Text
      accessible={true}
      onPress={[Function]}
      style={
        Object {
          "color": "#2089DC",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginVertical": 2,
          "textDecorationLine": "underline",
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 14,
          "marginVertical": 2,
        }
      }
    >
      

    </Text>
  </View>
</View>
`;

exports[`Resources renders properly in spanish 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "padding": 20,
      }
    }
  >
    <Text
      accessible={true}
      onPress={[Function]}
      style={
        Object {
          "color": "#2089DC",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginVertical": 2,
          "textDecorationLine": "underline",
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 14,
          "marginVertical": 2,
        }
      }
    >
      

    </Text>
  </View>
</View>
`;

exports[`Resources renders two resources 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "padding": 20,
      }
    }
  >
    <Text
      accessibilityHint="An Accessiblity Hint 1"
      accessibilityLabel="An Accessibility Label 1"
      accessible={true}
      onPress={[Function]}
      style={
        Object {
          "color": "#2089DC",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginVertical": 2,
          "textDecorationLine": "underline",
        }
      }
    >
      Resource Title 1
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
          "marginVertical": 2,
        }
      }
    >
      

      Resource Description 1
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "padding": 20,
      }
    }
  >
    <Text
      accessibilityHint="An accessiblity hint 2"
      accessibilityLabel="An accessibility label 2"
      accessible={true}
      onPress={[Function]}
      style={
        Object {
          "color": "#2089DC",
          "fontSize": 16,
          "fontWeight": "bold",
          "marginVertical": 2,
          "textDecorationLine": "underline",
        }
      }
    >
      Resource Title 2
    </Text>
    <Text
      style={
        Object {
          "fontSize": 14,
          "marginVertical": 2,
        }
      }
    >
      

      Resource Description 2
    </Text>
  </View>
</View>
`;
